### PR TITLE
Adds detailed documentation.

### PR DESCRIPTION
**Pull request by michaelgoin**
## Links

* Closes: https://github.com/newrelic/newrelic-node-apollo-server-plugin/issues/5

## Details

Adds detailed documentation including the new configuration and details on how we are naming things, metrics and how to visualize those.

Approach attempted is surfacing up some details on the README and then pointing to specific markdown pages for more in-depth details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,7 +66,116 @@ The New Relic plugin is known to work with the following Apollo Server modules:
...
+```
+FROM Metric
+SELECT average(newrelic.timeslice.value) * 1000 WHERE appName = '[YOUR APP NAME]' WITH METRIC_FORMAT 'GraphQL/resolve/ApolloServer/{field}' FACET field LIMIT 10
+```
+
+For more information on metrics and some recommended visualizations, please see the [metrics documentation](../docs/metrics.md).
+
+### Segments and Spans
+
+[Segments and Spans](./segments-and-spans.md)
```
Comment (carlo-808) [inline README.md:150]:
```suggestion
[Segments and Spans](./docs/segments-and-spans.md)
```

File: README.md [hunk 2]
```diff
@@ -31,7 +66,116 @@ The New Relic plugin is known to work with the following Apollo Server modules:
...
+}
+```
+
+`post /query/<anonymous>/libraries.books.author.name`
+
+For more information on how transactions are named, including how query errors may impact naming, please see the [transaction documentation](../docs/transactions.md).
+
+### Metrics
+
+[Metrics Documentation](../docs/metrics.md)
```
Comment (carlo-808) [inline README.md:117]:
```suggestion
[Metrics Documentation](./docs/metrics.md)
```

File: README.md [hunk 3]
```diff
@@ -4,7 +4,7 @@
 
 New Relic's official Apollo Server plugin for use with the [Node.js agent](https://github.com/newrelic/node-newrelic).
```
Comment (carlo-808) [inline README.md:6]: Should we be more explicit about how the graphql server needs to have the agent required?

Comment (michaelgoin) [reply README.md:6]: Wouldn't hurt

**Review by carlo-808 — APPROVED** [review]
👍

**Pull request merged**